build/charts/theia/values.yaml

```diff
@@ -0,0 +1,101 @@
+clickhouse:
+  # -- Container image to use for the ClickHouse.
+  image: 
+    repository: "projects.registry.vmware.com/antrea/theia-clickhouse-server"
+    pullPolicy: "IfNotPresent"
+    tag: "21.11"
+  monitor:
+    # -- Determine whether to run a monitor to periodically check the ClickHouse
+    # memory usage and clean data.
+    enable: true
+    # -- The storage percentage at which the monitor starts to delete old records.
+    # Vary from 0 to 1.
+    threshold: 0.5
+    # -- The percentage of records in ClickHouse that will be deleted when the
+    # storage grows above threshold. Vary from 0 to 1.
+    deletePercentage: 0.5
+    # -- The time interval between two round of monitoring. Can be a plain integer using
+    # one of these unit suffixes ns, us (or µs), ms, s, m, h.
+    execInterval: "1m"
+    # -- The number of rounds for the monitor to stop after a deletion to wait for
+    # the ClickHouse MergeTree Engine to release memory.
+    skipRoundsNum: 3
+    # -- Container image to use for the ClickHouse Monitor.
+    image:
+      repository: "projects.registry.vmware.com/antrea/theia-clickhouse-monitor"
+      pullPolicy: "IfNotPresent"
+      tag: "latest"
+  secret:
+    # -- ClickHouse username. It will be stored in a secret.
+    username: "clickhouse_operator"
+    # -- ClickHouse password. It will be stored in a secret.
+    password: "clickhouse_operator_password"
+  service:
+    # -- The type of service exposes ClickHouse. It can be one of ClusterIP, NodePort or LoadBalancer.
+    type: ClusterIP
+    # -- TCP port number for the ClickHouse service.
+    tcpPort: 9000
+    # -- TCP port number for the ClickHouse service.
+    httpPort: 8123
+  # -- ClickHouse storage size. Can be a plain integer or as a fixed-point
+  # number using one of these quantity suffixes: E, P, T, G, M, K. Or the
+  # power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki.
+  storageSize: "8Gi"
+  # -- Time to live for data in the ClickHouse. Can be a plain integer using
+  # one of these unit suffixes SECOND, MINUTE, HOUR, DAY, WEEK, MONTH, QUARTER,
+  # YEAR.
+  ttl: 1 HOUR
+  persistentVolume:
```

Comment:
I think this whole section can be a bit confusing, because it conflates some things, like PV type & storage class provisioner, and doesn't clearly distinguish between manual volume provisioning and dynamic volume provisioning.

It seems that there is also no option for users to provide a custom PersistentVolume (if they don't want to use dynamic provisioning).

```yaml
storage:
  size: "8Gi"
  createPersistentVolume:
    # can be set to "Local" or "NFS"
    type: ""
    local:
      path: "/data"
    nfs:
      host: ""
      path: ""
  # ignored if persistentVolume.type is non-empty
  persistentVolumeClaimSpec: {}
    # storageClassName: ""
    # volumeName: 
```

By default, memory storage is used.

To use a local volume created by us, one can set:
```
storage.createPersistentVolume.type = "Local"
```

To use a custom PersistentVolume, one can set:
```
storage.persistentVolumeClaimSpec:
  storageClassName: ""
  volumeName: "<my-pv>"
```

To dynamically provision a PersistentVolume, one can set:
```
storage.persistentVolumeClaimSpec
  storageClassName: <"my-storage-class">
```

Let me know what you think of that solution

Reply:
I think PersistentVolumeClaim does not accept PV name as a specification. As introduced in [K8S doc](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1), I think it means

- If storageClassName is specified, PVC will look for PV in the cluster with that class
- If storageClassName is set to "", PVC will look for PV with no class

From my experience, if users don't want to use dynamic provisioning, they could set StorageClass to "" or create a StorageClass with `kubernetes.io/no-provisioner` as provisioner, and make sure the corresponding PV is created manually. 

We can still adopt the solution, with the `storage.persistentVolumeClaimSpec` only expecting `storageClassName`. Memory storage will be used when `storage. createPersistentVolume.type` is "" and `storage.persistentVolumeClaimSpec` is empty. But I'm not sure whether this change is necessary as it does not distinguish between manual volume provisioning and dynamic volume provisioning either.

Reply:
There are 3 cases for storageClassName: set, omitted and set to empty
I do think it is possible to manually bind to a PV with the `volumeName` field: https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reserving-a-persistentvolume

Reply:
I missed this. Thanks! The storage section is refactored now.